docs/BaremetalHost_ProvisioningState.dot

```diff
@@ -51,11 +51,13 @@ digraph BaremetalHost {
     Provisioned -> Deprovisioning [label="!DeletionTimestamp.IsZero()"]
 
     ExternallyProvisioned [shape=doublecircle]
-    ExternallyProvisioned -> Deleting [label="!DeletionTimestamp.IsZero()"]
+    ExternallyProvisioned -> PoweringOffBeforeDelete [label="!DeletionTimestamp.IsZero()"]
```

Comment:
People have complained about all the `Deleting3 [shape=point]` stuff making it unclear that the host goes into the Deleting state no matter where it was when deleted.
I wonder if this is the point where it finally becomes completely untenable to hide this info for the sake of saving a few lines on the diagram. How bad does it look if we include all the edges?

Reply:
Something like this:

![BaremetalHost_ProvisioningState](https://github.com/metal3-io/baremetal-operator/assets/206357/489dfc78-a325-4e1f-a603-e254af0e8236)

Reply:
:thinking: that's... not outstanding, but maybe not so bad that it's worth the ongoing confusion to avoid it. WDYT?
We don't actually go to PoweringOffBeforeDelete from Unmanaged or Registering, do we?

Reply:
> We don't actually go to PoweringOffBeforeDelete from Unmanaged or Registering, do we?

We actually [do](https://github.com/metal3-io/baremetal-operator/blob/5a01d08058d63e1e98b4ebe5e6b54c6f50d1837e/controllers/metal3.io/host_state_machine.go#L226-L227), but the (inevitable) error is [handled](https://github.com/metal3-io/baremetal-operator/blob/5a01d08058d63e1e98b4ebe5e6b54c6f50d1837e/controllers/metal3.io/host_state_machine.go#L588-L593), so I guess not technically a bug even though it is a bit messy for the user.

Reply:
OK, I pushed the diagram changes.  I think it's reasonably clear.